### PR TITLE
Migrate FromSourceAliasRewriter to PartiqlAst.VisitorTransform

### PR DESCRIPTION
**Pull request by alancai98**
~#298 PR should be merged into `visitor-transforms` first (contains changes to metas and adds the `VisitorTransform`-related classes).~ (has already been merged, so this can be looked at)

Migrates `FromSourceAliasRewriter` to `PartiqlAst.VisitorTransform`. Conveniently, there were previous tests for `FromSourceAliasRewriter`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lang/src/org/partiql/lang/ast/ExprNodeToStatement.kt [hunk 1]
```diff
@@ -335,3 +336,7 @@ private fun DataType.toAstType(): PartiqlAst.Type {
         }
     }
 }
+
+
+private fun SymbolicName.toPrimitive(): SymbolPrimitive =
+    SymbolPrimitive(this.name, this.metas.toIonElementMetaContainer())
```
Comment (therapon) [inline lang/src/org/partiql/lang/ast/ExprNodeToStatement.kt:342]: We had this exact function in the other PR, could we keep one implementation and reuse?

File: lang/src/org/partiql/lang/ast/passes/RewriterTransformBridge.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.partiql.lang.ast.passes
```
Comment (therapon) [inline lang/src/org/partiql/lang/ast/passes/RewriterTransformBridge.kt:1]: Same comment about duplicate code here. Am I missing something or is this due to the other PR not being committed and we will clean the 2 PRs accordingly before commit to master?

File: lang/src/org/partiql/lang/eval/CompileOptions.kt [hunk 3]
```diff
@@ -49,13 +50,13 @@ enum class ProjectionIterationBehavior {
...
+        override fun createRewriter(ion: IonSystem) = basicRewriters(ion)
     },
     NONE {
-        override fun createRewriter() = IDENTITY_REWRITER
+        override fun createRewriter(ion: IonSystem) = IDENTITY_REWRITER
     };
 
-    internal abstract fun createRewriter(): AstRewriter
+    internal abstract fun createRewriter(ion: IonSystem): AstRewriter
 }
```
Comment (therapon) [inline lang/src/org/partiql/lang/eval/CompileOptions.kt:61]: These seem duplicates as well

**Review by therapon — COMMENTED** [review]
I think I am missing something with regards to the PRs on visitors. I see some duplicate code and I am not sure if that is needed to make the CR and Build to pass and we will amend once the 1st of the 2 PRs gets committed.

File: lang/src/org/partiql/lang/ast/ExprNodeToStatement.kt (hunk 1 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/ast/ExprNodeToStatement.kt:342]: Sorry for the confusion. The other PR (#298) needs to be merged first into the `visitor-transforms` feature branch.

File: lang/src/org/partiql/lang/ast/passes/RewriterTransformBridge.kt (hunk 2 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/ast/passes/RewriterTransformBridge.kt:1]: Sorry for the confusion again. The other visitor transform PR (#298) should be merged first into the `visitor-transform` feature branch. Once all the visitor transform PIG migrations are done (i.e. committed into the `visitor-transform` feature branch), I believe there will just be one commit into master.

**Comment by alancai98** [comment]
Sorry for any confusion. This PR should be looked at once the first Rewriter to VisitorTransform PR (#298) is merged into the `visitor-transform` feature branch.

In the meantime, the most recent commit on this PR ([94a58d5](https://github.com/partiql/partiql-lang-kotlin/pull/302/commits/94a58d500aad39198f7da1d31942c1b27b92b551)) has all the relevant changes for this task.

Not too sure why the travis-ci test fails though.

**Review by therapon — APPROVED** [review]

File: lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+class FromSourceAliasVisitorTransform : PartiqlAst.VisitorTransform() {
+
+    private class InnerFromSourceAliasVisitorTransform : PartiqlAst.VisitorTransform() {
+        private var fromSourceCounter = 0
+
+        override fun transformFromSourceScan_asAlias(node: PartiqlAst.FromSource.Scan): SymbolPrimitive? {
+            val thisFromSourceIndex = fromSourceCounter++
+            return node.asAlias ?:
+                transformSymbolPrimitive(
+                    SymbolPrimitive(node.expr.extractColumnAlias(thisFromSourceIndex), node.extractSourceLocation()))
```
Comment (dlurton) [inline lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt:27]: I don't believe it is necessary to transform the newly constructed `SymbolPrimitive` so I think you can do away with the call to `transformSymbolPrimitive`.

File: lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        override fun transformFromSourceUnpivot_asAlias(node: PartiqlAst.FromSource.Unpivot): SymbolPrimitive? {
+            val thisFromSourceIndex = fromSourceCounter++
+            return node.asAlias ?:
+                transformSymbolPrimitive(
```
Comment (dlurton) [inline lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt:33]: same

**Review by dlurton — CHANGES_REQUESTED** [review]
One minor change, otherwise LGTM.

File: lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt (hunk 4 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt]: Got it, removed.

File: lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt (hunk 5 above)
Comment (alancai98) [reply lang/src/org/partiql/lang/eval/visitors/FromSourceAliasVisitorTransform.kt]: Removed.

**Review by dlurton — APPROVED** [review]

**Pull request merged**